### PR TITLE
[VA-IIR 379] Make Academic Year Automatic and Add to SearchResults Container

### PR DESCRIPTION
**Pull request by john-rodriguez-adhocteam**
## Summary

- The following code updates the [Yellow Ribbon Search tool's](https://www.va.gov/education/yellow-ribbon-participating-schools/) SearchResults container to include academic year based on VA-IIR UX feedback as well as makes the year automatic.
- I work for the VA-IIR (a.k.a., "Iterate, Innovate, and Run") contractor team under Oddball (primary) and Ad Hoc.

## Related issue(s)

- [VA-IIR Zenhub ticket 379](https://app.zenhub.com/workspaces/va-iir-6508c0bd79e64e0fb5855caf/issues/gh/department-of-veterans-affairs/va-iir/379)

## Testing done

- The function used to automate the academic year is tested as part of the helper function tests.
- Also conducted a visual check that the year is correct and now also shows above the search results.

## Screenshots

| Before | After |
| ------ | ------ |
Yellow Ribbon SearchResults message saying "Displaying 1–10 of 85 results" | Yellow Ribbon SearchResults message saying "Showing 1–10 of 31 schools for academic year 2023 to 2024."

## Acceptance criteria

- [x] The academic year is displayed as part of the results portion of the page.
- [x] Academic year is now automatically updated.

### CODE REVIEW

File: src/applications/yellow-ribbon/components/SearchResultsPage/index.js [hunk 1]
```diff
@@ -34,13 +34,14 @@ export const SearchResultsPage = ({
...
           Information for participating schools is for the current academic
-          year, {CURRENT_SCHOOL_YEAR}.
+          year, {getCurrentAcademicYear()}.
         </p>
       </div>
 
       {/* Search Form */}
       <div className="vads-l-col--12 medium-screen:vads-l-col--3">
         {/* Toggle Mobile Form */}
+        {/* eslint-disable-next-line @department-of-veterans-affairs/prefer-button-component */}
```
[the remark this answers is not in the document]
Comment (john-rodriguez-adhocteam) [reply src/applications/yellow-ribbon/components/SearchResultsPage/index.js:44]: Outside the scope of the ticket.

File: src/applications/yellow-ribbon/containers/SearchResults/index.jsx [hunk 2]
```diff
@@ -1,3 +1,6 @@
+/* eslint-disable react/sort-prop-types */
```
[the remark this answers is not in the document]
Comment (john-rodriguez-adhocteam) [reply src/applications/yellow-ribbon/containers/SearchResults/index.jsx:1]: We will be creating tickets to address refactors/updates discovered in this file while making these changes.

File: src/applications/yellow-ribbon/components/SearchResultsPage/index.js [hunk 3]
```diff
@@ -93,12 +94,10 @@ export const SearchResultsPage = ({
...
 
 SearchResultsPage.propTypes = {
-  // From mapStateToProps.
   hasFetchedOnce: PropTypes.bool.isRequired,
   showMobileForm: PropTypes.bool.isRequired,
-  totalResults: PropTypes.number,
-  // From mapDispatchToProps.
   toggleShowMobileForm: PropTypes.func.isRequired,
+  totalResults: PropTypes.number,
 };
```
Comment (nicksayre) [inline src/applications/yellow-ribbon/components/SearchResultsPage/index.js:101]: I appreciate the alphabetization!

Comment (john-rodriguez-adhocteam) [reply src/applications/yellow-ribbon/components/SearchResultsPage/index.js:101]: Would love to take credit, but it's required by the checks lol

File: src/applications/yellow-ribbon/helpers/index.unit.spec.js [hunk 4]
```diff
@@ -89,4 +94,17 @@ describe('Yellow Ribbon helpers', () => {
       });
     });
   });
+
+  describe('getCurrentAcademicYear', () => {
+    xit('returns Academic Year "2023 to 2024" if the current date is before August 2024', () => {
```
Comment (nicksayre) [inline src/applications/yellow-ribbon/helpers/index.unit.spec.js:99]:
```suggestion
    it('returns Academic Year "2023 to 2024" if the current date is before August 2024', () => {
```

**Review by nicksayre — CHANGES_REQUESTED** [review]
Looks good other than a typo on the added test.

**Review by nicksayre — APPROVED** [review]

**Review by Andrew565 — APPROVED** [review]

**Pull request merged**